lib/Model/Type.cpp

```diff
@@ -770,28 +770,20 @@ Type::trySize(VerifyHelper &VH) const {
     rc_return MaybeSize;
 
   // This code assumes that the type T is well formed.
-  uint64_t Size;
+  uint64_t Size = 0;
 
   switch (Kind()) {
-  case TypeKind::Invalid:
-    rc_return std::nullopt;
-
   case TypeKind::RawFunctionType:
   case TypeKind::CABIFunctionType:
     // Function prototypes have no size
-    Size = 0;
-    break;
+    rc_return std::nullopt;
```

Comment:
I'm planning to overhaul `size` and `trySize` (I've to figure out we we need both).

Did you already figure out why sometimes we return 0 and sometimes `std::nullopt` in `size`/`trySize`?
If not, and this doesn't break any code, I'd ask you to leave it out.

Reply:
I did (all in effort to making sure `size()` and `alignment()` stay in sync).

In short, `trySize` used to return `std::nullopt` if it cannot figure size in principle (for example, there's a loop in our type system), but `0` for types without the size (like `void` and function types).
Then `size` call `trySize` internally, asserts on its `std::nullopt` and upgrades its `0` into a `std::nullopt` of its own.

Reply:
OK, this makes sense! Can you make sure it's documented in Doxygen of the methods?

Reply:
Sounds good.

Reply:
Is this about what you had in mind?
```cpp
  /// Recursively computes the size of the type.
  ///
  /// It asserts in cases where the size cannot be computed, for example, when
  /// the type system loops and the type's size depends on the type itself.
  ///
  /// \returns `std::nullopt` if the type does not have size (for example, it's
  ///          a `void` primitive or a function type), the size in bytes
  ///          otherwise.
  std::optional<uint64_t> size() const debug_function;
  std::optional<uint64_t> size(VerifyHelper &VH) const;

  /// Tries to recursively compute the size of the type.
  ///
  /// It asserts in cases where the size cannot be computed, for example, when
  /// the type system loops and the type's size depends on the type itself.
  ///
  /// \returns `std::nullopt` if the size cannot be computed, for example, when
  ///          the type system loops and the type's size depends on the type
  ///          itself, the size in bytes otherwise.
  std::optional<uint64_t> trySize() const debug_function;
  RecursiveCoroutine<std::optional<uint64_t>> trySize(VerifyHelper &VH) const;
```

Reply:
The commit is `Model: document `model::Type::size()`.